### ISSUE TITLE
Roadmap of improvement

### ISSUE THREAD

**Issue by DavideCarvalho** [georgemandis/konami-js#44]
I saw that you wanted some help to improve this lib, and i'd like to help you with it! So, I wanted to know if you have some roadmap of improvement.

Some things that caught my eye were:
- Lack of eslint
- No babel for es6
- No bundler for libraries, like Rollup
- Bower (do you intend on keeping bower support? Not even bower supports it)
- Do you want some new features? Maybe listing would be nice so people can help

**Comment by georgemandis** [comment]
Hi @DavideCarvalho —

Thank you very much for reaching out! I love the idea of developing a roadmap and definitely could use some help from the community. I've started a [project board for Konami-JS 2.0](https://github.com/snaptortoise/konami-js/projects/1). 

I'd like to come up with a list of improvements and features. I've created some labels for the issues here and provided descriptions.

So far I feel most confident on these items:

- Integrating Babel to accommodate transpiling for older browsers (#47)
- Incorporating an ESLint configuration (#45)
- Dropping Bower support (#46)

The bundler issue is also something I definitely want addressed, but I'm not sure what direction to go. Should I just choose one (webpack, rollup, etc) and go with it?  Should I add configuration files trying  to accommodate as many as I can? I have surprisingly few strong opinions on this. My concern with this is always choosing something that gets phased out down the line (like Bower). Maybe to some degree that's inevitable... but it bothers me :)

I'm opening a separate issue (#48) to discuss the best approach to this.

As for new features, I'm open to suggestions! Currently the project listens for and triggers "easter eggs" in response to keyboard or touchscreen input. An "easter egg" can be any bit of JavaScript code but defaults to loading a new webpage in the browser. I don't feel a pressing need to add new features here, but would love to hear other people's thoughts. I've also opened an issue to discuss this (#49).
